services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/DynamoDbAsyncIndex.java

```diff
@@ -27,28 +27,15 @@
  * @param <T> The type of the modelled object.
  */
 @SdkPublicApi
-public interface AsyncMappedIndex<T> {
-    /**
-     * Executes a command that is expected to return a single data item against the database with the context of the
-     * specific table and secondary index this object is linked to.
-     *
-     * @param operationToPerform The operation to be performed in the context of the secondary index.
-     * @param <R> The expected return type from the operation. This is typically inferred by the compiler.
-     * @return A {@link CompletableFuture} of the result of the operation being executed. The documentation on the
-     * operation itself should have more information.
-     */
-    <R> CompletableFuture<R> execute(IndexOperation<T, ?, ?, R> operationToPerform);
+public interface DynamoDbAsyncIndex<T> {
 
-    /**
-     * Executes a command that is expected to return a paginated list of data items against the database with the
-     * context of the specific table and secondary index this object is linked to.
-     *
-     * @param operationToPerform The operation to be performed in the context of the secondary index.
-     * @param <R> The expected return type from the operation. This is typically inferred by the compiler.
-     * @return An {@link SdkPublisher} that will publish successive pages of result data items to any subscriber with
-     * demand for them.
-     */
-    <R> SdkPublisher<R> execute(PaginatedIndexOperation<T, ?, ?, R> operationToPerform);
+    default SdkPublisher<Page<T>> query(QueryEnhancedRequest request) {
+        throw new UnsupportedOperationException();
```

Comment:
Why the need for defaults here?

Reply:
For supporting new operations in the future, I think, and now for consistency. It's in the design doc.